### PR TITLE
Ericvoit patch 4

### PR DESCRIPTION
**Pull request by ericvoit**
Post IETF 111 comments.

### CODE REVIEW

File: draft-voit-rats-attestation-results.md [hunk 1]
```diff
@@ -273,55 +278,233 @@ For more on this, see {{freshness-section}}.
...
+
+   Reason: a plethora of similar Trustworthiness Claims will result in divergent choices made on which to support between different Verifiers.  This would place a lot of complexity in the Relying Party as it would be up to the Relying Party (and its policy language) to enable normalization across rich but incompatible Verifier object definitions.
+
+2. Each Trustworthiness Claim should enumerate only the specific states that could viably result in a different outcome after the Policy for Attestation Results has been applied.   
+
+   Reason: by explicitly disallowing the standardization of enumerated states which cannot easily be connected to a use case, we avoid forcing implementers from making incompatible guesses on what these states might mean.  
+
+3. Verifier and RP developers need explicit definitions of each state in order to accomplish the goals of (1) and (2).  
+
+   Reason: without such guidance, the Verifier will append plenty of raw supporting info.  The  relieves the Verifier of making the hard decisions.  Of course, this raw info will be mostly non-interpretable and therefore non-actionable by the Relying Party.
```
Comment (thomas-fossati) [inline draft-voit-rats-attestation-results.md]: typo

```suggestion
   Reason: without such guidance, the Verifier will append plenty of raw supporting info.  This relieves the Verifier of making the hard decisions.  Of course, this raw info will be mostly non-interpretable and therefore non-actionable by the Relying Party.
```

File: draft-voit-rats-attestation-results.md [hunk 2]
```diff
@@ -273,55 +278,233 @@ For more on this, see {{freshness-section}}.
...
+
+4. Support standards and non-standard extensibility for (1) and (2).  
+
+   Reason: The need is obvious.  Verifier generated claims should be vetted by the full RATS working group, rather than being in a separately maintained Attester Claims list.  This will keep a tight lid on extensions which must be considered by the Relying Party's policy language.
```
Comment (thomas-fossati) [inline draft-voit-rats-attestation-results.md:305]: I am not sure I understand what we mean by "Attester Claims list"?

Comment (thomas-fossati) [reply draft-voit-rats-attestation-results.md:305]: Another comment I have on this bit is that the stated rationale (the "Reason" part) does seem to cover the non-standard extensibility part.

File: draft-voit-rats-attestation-results.md [hunk 3]
```diff
@@ -273,55 +278,233 @@ For more on this, see {{freshness-section}}.
...
+
+* Values 1-31: A standards enumerated reason for affirming.
+* Values -1 to -32: A non-standard reason for affirming.
+
+Warning: The Verifier warns about this aspect of trustworthiness.
+
+* Values 32-63: A standards enumerated reason for the warning.
+* Values -33 to -64: A non-standard reason for the warning.
+
+Contraindicated: The Verifier does not support this aspect of trustworthiness.
```
Comment (thomas-fossati) [inline draft-voit-rats-attestation-results.md:324]: Non-native comment: I find the use of the verb "support" a bit confusing as I map it automatically to the idea of "supporting a feature".  I'd use "accept" or "successfully verify" instead.

File: draft-voit-rats-attestation-results.md [hunk 4]
```diff
@@ -273,55 +278,233 @@ For more on this, see {{freshness-section}}.
...
+
+1. If applicable, a Relying Party MUST assign a standardized code from the Contraindicated tier.   
+2. Else if applicable, a Relying Party MUST assign a non-standardized code from the Contraindicated tier.    
+3. Else if applicable, a Relying Party MUST assign a standardized code from the Warning tier.  
+4. Else if applicable, a Relying Party MUST assign a non-standardized code from the Warning tier.
+5. Else if applicable, a Relying Party MUST assign a standardized code from the Affirming tier.    
+6. Else if applicable, a Relying Party MUST assign a non-standardized code from the Affirming tier. 
+7. Else a Relying Party MAY assign a 0.     
+
+The standardized Trustworthiness Claim values listed below have been design so there is no overlap within a trustworthiness tier.
```
Comment (thomas-fossati) [inline draft-voit-rats-attestation-results.md]: typo

```suggestion
The standardized Trustworthiness Claim values listed below have been designed so there is no overlap within a trustworthiness tier. 
```

File: draft-voit-rats-attestation-results.md [hunk 5]
```diff
@@ -273,55 +278,233 @@ For more on this, see {{freshness-section}}.
...
+* Value 0: Note: this should always be always treated equivalently by the Relying Party as no claim being made. I.e., the RP's Appraisal Policy for Attestation Results SHOULD NOT make any distinction between a Trustworthiness Claim with enumeration '0', and no Trustworthiness Claim being provided.
+
+This enumerated encoding listed above will simplify the Appraisal Policy for Attestation Results.
+Such a policies may be as simple as saying that a specific Verifier has recently asserted Trustworthiness Claims, all of which are Affirming.
+
+### Assigning a Trustworthiness Claim value
+
+In order to simplify design, only a single encoded value is asserted for any Trustworthiness Claim within a Trustworthiness Vector using the following process.  
+
+1. If applicable, a Relying Party MUST assign a standardized code from the Contraindicated tier.
```
Comment (thomas-fossati) [inline draft-voit-rats-attestation-results.md:340]: I am a bit confused here: why is the Relying Party the subject of the sentences?  I would have expected it to be the Verifier.

File: draft-voit-rats-attestation-results.md [hunk 6]
```diff
@@ -273,55 +278,233 @@ For more on this, see {{freshness-section}}.
...
+   : the Attester encrypts all secrets in persistent storage via using keys which are never visible outside an HSM or the Trusted Execution Environment hardware.
+  
+   32:
+   : the Attester encrypts all persistently stored secrets, but without using hardware backed keys
+
+   64:
+   : There are persistent secrets which are stored unencrypted in an Attester.
+     
+
+Additional Trustworthiness Claims may be defined in subsequent documents.
```
Comment (thomas-fossati) [inline draft-voit-rats-attestation-results.md:486]: Question: what about the values for the claims we define in this document?  Is it possible for future documents to extend them, or are they fixed forever?  If the former, how does an RP deals with unknown values?

**Review by thomas-fossati — COMMENTED** [review]
Eric, thanks for an excellent set of edits!  (I have left a few comments inline.)

File: draft-voit-rats-attestation-results.md (hunk 2 above)
Comment (ericvoit) [reply draft-voit-rats-attestation-results.md:305]: > I am not sure I understand what we mean by "Attester Claims list"?

This is the set of Trustworthiness Claims about an Attester.   Laurence has asserted that certain people might be able to create and push incremental EAT claims right to an authoritative repository .  That shouldn't be the case here.  I will tweak the text to clarify.

Comment (ericvoit) [reply draft-voit-rats-attestation-results.md:305]: > Another comment I have on this bit is that the stated rationale (the "Reason" part) does seem to cover the non-standard extensibility part.

Will add a brief sentence.

File: draft-voit-rats-attestation-results.md (hunk 5 above)
Comment (ericvoit) [reply draft-voit-rats-attestation-results.md:340]: Oooops :-)    Thanks for the catch.

File: draft-voit-rats-attestation-results.md (hunk 3 above)
Comment (ericvoit) [reply draft-voit-rats-attestation-results.md:324]: will make the definition "The Verifier asserts the Attester is explicitly untrustworthy in regard to this aspect"

File: draft-voit-rats-attestation-results.md (hunk 6 above)
Comment (ericvoit) [reply draft-voit-rats-attestation-results.md:486]: It is possible for future documents to extend them.   As for unknown values, they will fall into the ranges for Affirming/Warning/Contraindicated.  So the RP can operate on generalities if it so chooses.   I will explicitly state this.

File: draft-voit-rats-attestation-results.md (hunk 2 above)
Comment (ericvoit) [reply draft-voit-rats-attestation-results.md:305]: New text will be posted.   Closing comment.

File: draft-voit-rats-attestation-results.md (hunk 3 above)
Comment (ericvoit) [reply draft-voit-rats-attestation-results.md:324]: New text will be posted.   Closing comment.

**Pull request merged**